dashboard/app/models/levels/blockly.rb

```diff
@@ -564,4 +564,30 @@ def shared_functions
       SharedBlocklyFunction.where(level_type: type).map(&:to_xml_fragment)
     end.join
   end
+
+  def get_localized_for_block_type(script_name, default_language_code, language_code)
+    translated_block_text = []
+    current_script = Script.find_by(name: script_name)
+    return translated_block_text if current_script.blank?
+
+    current_script_levels = current_script.script_levels
+    if current_script_levels.present?
+      current_script_levels.map do |script_level|
+        level = script_level.level
+        next if level.blank?  #consider next unless level.present
+        level_shared_blocks = level.shared_blocks
+        next if level_shared_blocks.blank?
+        level_shared_blocks.map do |sharedblock|
+          next if sharedblock[:config].blank?
+          block_text = sharedblock[:config]['blockText']  #this is only for text, there is also options
+          if block_text.present?
+            I18n.locale = language_code
+            translated_block_text << I18n.t('data.blocks.Dancelab_atTimestamp.text')  #need to change to account for craft
+          end
+        end
+      end
+    end
+    I18n.locale = default_language_code  #default to english if string isn't translated
```

Comment:
I would very strongly encourage against this pattern; it should never be possible to accidentally change the i18n locale for the rest of the request by passing in bad values to this function.

Depending on how this method is intended to be used, I would recommend avoiding modifying `i18n.locale` at all, and instead just serve content for whichever language it is already set to. If there is a specific reason why this is built to accept the target language as an argument, then what you want to do is something like:

```ruby
def get_localized_for_block_type(script_name, language_code)
  # save the locale as it was before the method was called, and only then change the locale
  original_locale = I18n.locale
  I18n.locale = language_code
  # all the other code here
  ...
  # finally, restore the original locale
  I18n.locale = original_locale
end
```

Reply:
This was my bad, when we were testing out the change in the console I was manually setting `I18n.locale` to see the result:

```ruby
[50] pry(main)> I18n.locale = 'es'
=> "es"
[51] pry(main)> I18n.t('data.blocks.Dancelab_atTimestamp.text')
=> "después de {TIMESTAMP} {UNIT}"
```

@nkiruka for the shipped version of this, we shouldn't need to modify `I18n.locale`.  We do this in one centralized place for all incoming requests:

https://github.com/code-dot-org/code-dot-org/blob/d44b0298adc9ee2453b79644c7463bac64349657/lib/cdo/rack/locale.rb#L14-L22

Reply:
A bit more info here: https://guides.rubyonrails.org/i18n.html#managing-the-locale-across-requests